### PR TITLE
Replaced Managed Instance icon on the Object Explorer

### PR DESCRIPTION
**Pull request by nemanja-milovancevic**
Object Explorer used to show the same icon for Managed Instance and SQL Server:
image

In this PR, I changed the default icon for Managed Instance:
image
image

PR includes both dark and light theme icons.

### CODE REVIEW

File: extensions/mssql/src/iconProvider.ts [hunk 1]
```diff
@@ -7,14 +7,18 @@ import * as azdata from 'azdata';
...
+const miIcon = "mssql:miIcon";
 
 export class MssqlIconProvider implements azdata.IconProvider {
 	public readonly providerId: string = constants.sqlProviderName;
 	public handle: number;
 	getConnectionIconId(connection: azdata.IConnectionProfile, serverInfo: azdata.ServerInfo): Thenable<string> {
 		let iconName: string = undefined;
 		if (connection.providerName === 'MSSQL') {
-			if (serverInfo.isCloud) {
+			if (serverInfo.engineEditionId === 8) {
```
Comment (kburtram) [inline extensions/mssql/src/iconProvider.ts:18]: Is it safe to assume this edition is always only MI?

**Review by kburtram — APPROVED** [review]
LGTM assuming edition uniquely identifies MI.

File: extensions/mssql/src/iconProvider.ts (hunk 1 above)
Comment (Charles-Gagnon) [inline extensions/mssql/src/iconProvider.ts:18]:
```suggestion
			if (serverInfo.engineEditionId === azdata.DatabaseEngineEdition.SqlManagedInstance) {
```

Comment (nemanja-milovancevic) [reply extensions/mssql/src/iconProvider.ts:18]: Yes, it's safe. Engine edition 8 means its Managed Instance. 
https://learn.microsoft.com/en-us/sql/t-sql/functions/serverproperty-transact-sql?view=sql-server-ver16

**Review by Charles-Gagnon — APPROVED** [review]

**Comment by Charles-Gagnon** [comment]
@nemanja-milovancevic Looks like the CLA check is stuck again...we'll wait a bit and then try refreshing to see if it's fixed itself by then.

**Comment by Charles-Gagnon** [comment]
@nemanja-milovancevic You can delete your branch since I got this in with the other PR

**Pull request closed without merging**